cdk/lib/stack.test.ts

```diff
@@ -1,10 +1,14 @@
 import { App, assertions } from "aws-cdk-lib";
-import { PinBoardStack } from "../stack";
+import { PinBoardStack } from "./stack";
 
 describe("PinBoardStack's", () => {
   it("generated CloudFormation matches the snapshot", () => {
     const app = new App();
-    const stack = new PinBoardStack(app, "PinBoardStack");
+    const stack = new PinBoardStack(app, "PinBoardStack-TEST", {
+      domainName: "pinboard.test.dev-gutools.co.uk",
+      stack: "workflow",
+      stage: "TEST",
+    });
```

Comment:
Just for my understanding, what does this do?

Reply:
This is the file which triggers the snapshot tests. And since the stage specific variables are passed as props to `PinboardStack` we supply test related ones in the test.

Snapshot tests allow us to see a more verbose/representative view of the infrastructure changes we're making, compared to the abstraction that is CDK, and further abstraction that is guCDK. E.g. one line of CDK can result in many lines of cloudformation. So we make the snapshot also part of VCS so in PR code review we can see not only the diff in CDK code but also the resulting diff in generated cloudformation.